src/main/java/org/gaul/s3proxy/AliasBlobStore.java

```diff
@@ -0,0 +1,297 @@
+        for (String key : properties.stringPropertyNames()) {
+            Matcher matcher = ALIAS_RE.matcher(key);
+            if (matcher.matches()) {
+                String virtualBucket = matcher.group("bucket");
+                String backendBucket = properties.getProperty(key);
+                checkArgument(
+                        !backendBuckets.contains(properties.getProperty(key)),
+                        "Backend bucket %s is aliased twice",
+                        properties.getProperty(key));
+                builder.put(virtualBucket, backendBucket);
+                backendBuckets.add(backendBucket);
+            }
+        }
+        return builder.build();
+    }
+
+    private String getContainer(String container) {
+        return this.aliases.getOrDefault(container, container);
+    }
+
+    @Override
+    public boolean createContainerInLocation(Location location,
+                                             String container) {
+        return this.delegate().createContainerInLocation(location,
+                getContainer(container));
+    }
+
+    @Override
+    public boolean createContainerInLocation(
+            Location location, String container,
+            CreateContainerOptions options) {
+        return delegate().createContainerInLocation(
+                location, getContainer(container), options);
+    }
+
+    @Override
+    public boolean containerExists(String container) {
+        return delegate().containerExists(getContainer(container));
+    }
+
+    @Override
+    public ContainerAccess getContainerAccess(String container) {
+        return delegate().getContainerAccess(getContainer(container));
+    }
+
+    @Override
+    public void setContainerAccess(String container,
+                                   ContainerAccess containerAccess) {
+        delegate().setContainerAccess(getContainer(container), containerAccess);
+    }
+
+    @Override
+    public PageSet<? extends StorageMetadata> list() {
+        PageSet<? extends StorageMetadata> upstream = this.delegate().list();
+        ImmutableList.Builder<StorageMetadata> results =
+                new ImmutableList.Builder<>();
+        for (StorageMetadata sm : upstream) {
+            if (aliases.containsValue(sm.getName())) {
+                MutableStorageMetadata bucketAlias =
+                        new MutableStorageMetadataImpl();
+                bucketAlias.setName(aliases.inverse().get(sm.getName()));
+                bucketAlias.setCreationDate(sm.getCreationDate());
+                bucketAlias.setETag(sm.getETag());
+                bucketAlias.setId(sm.getProviderId());
+                bucketAlias.setLastModified(sm.getLastModified());
+                bucketAlias.setLocation(sm.getLocation());
+                bucketAlias.setSize(sm.getSize());
+                bucketAlias.setTier(sm.getTier());
+                bucketAlias.setType(sm.getType());
+                bucketAlias.setUri(sm.getUri());
```

Comment:
Might want to add a TODO that the Uri probably points to the wrong thing.  This shouldn't affect S3Proxy though.

Reply:
Done